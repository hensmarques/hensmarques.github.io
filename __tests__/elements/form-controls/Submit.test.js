import React from 'react';
import Submit from '../../../src/elements/form-controls/Submit';

describe('<Submit/>', () => {
  it('renders with default props and matches the snapshot', () => {
    const wrapper = shallow(<Submit />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the props correctly', () => {
    const mockProps = {
      className: 'mockClass',
      value: 'Click Me',
      onSubmit: jest.fn(),
      onFocus: jest.fn(),
      onBlur: jest.fn(),
      formValid: true
    };
    const wrapper = shallow(<Submit {...mockProps} />);
    expect(wrapper.hasClass(mockProps.className)).toBe(true);
    expect(wrapper.find('input').hasClass(mockProps.className)).toBe(true);
    const inputProps = wrapper.find('input').props();
    expect(inputProps.value).toBe(mockProps.value);
    expect(inputProps.onFocus).toBe(mockProps.onFocus);
    expect(inputProps.onBlur).toBe(mockProps.onBlur);
    expect(inputProps.disabled).toBe(!mockProps.formValid);
  });

  it('fires onSubmit when component is clicked', () => {
    const mockProps = {
      onSubmit: jest.fn()
    };
    const wrapper = shallow(<Submit {...mockProps} />);
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.find('input').simulate('click', mockEvent);
    expect(mockProps.onSubmit).toHaveBeenCalledWith(mockEvent);
    expect(mockEvent.preventDefault).toHaveBeenCalled();
  });
});
