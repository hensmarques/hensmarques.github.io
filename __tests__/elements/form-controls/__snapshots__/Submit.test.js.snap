// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Submit/> renders with default props and matches the snapshot 1`] = `
<div
  className=""
>
  <input
    className="form-field submit-button "
    disabled={false}
    onClick={[Function]}
    type="submit"
    value="Submit"
  />
</div>
`;
